---
title: "Tides and Nearshore eDNA"
author: "Kelly, Gallego, and Jacobs-Palmer"
date: "Nov 2017"
output:
  pdf_document: 
    fig_caption: yes
    keep_tex: yes
    template: peerj.sty
  bibliography: TidesBibliography.bib
  html_document:
    fig_caption: yes
    force_captions: yes
    number_sections: no
    theme: readable
---

```{r setup, echo =FALSE, echo=F, warning=F, error=F, include=FALSE}

knitr::opts_chunk$set(
  comment = "#>",
  collapse = TRUE,
  warning = FALSE,
  message = FALSE,
  echo = FALSE,
  fig.align="center",
  fig.width = 6,
  fig.height = 4
)
options("max.print"=10000)
options(scipen = 0, digits = 4)
knitr::knit_hooks$set(inline = function(x) {
      if(is.numeric(x)){
          return(prettyNum(x, big.mark=","))
      }else{
          return(x)
       }
   })
```

```{r loadLibraries_etc, echo=F, error=F, message=FALSE, warning=F, include=FALSE}
##########LOAD LIBRARIES and FUNCTIONS#############
#How to check if you have all needed packages installed, introducing: uptodate.R
source("uptodate.R")
libs=c("vegan","plotrix","MASS","data.table", "gdata","lattice","plyr", "gridExtra", "readxl", "Biostrings", "ggmap", "xtable", "pander","RColorBrewer", "ggfittext", "treemapify", "kableExtra", "knitr", "tidyverse", "RCurl")
uptodate(libs)
lapply(libs, require, character.only = TRUE)

multiplot.script<-getURL("https://raw.githubusercontent.com/larmarange/JLutils/master/R/multiplot.R", ssl.verifypeer = FALSE); eval(parse(text = multiplot.script))  #ggplot multiplot function from github

bit=function(x){print(x[c(1:10),c(1:10)]); print(dim(x))}  #shows top-left corner of a data.frame/matrix; useful for large datasets
colMax <- function(data) sapply(data, max, na.rm = TRUE)
Col2RN<-function(df, x){row.names(df)<-df[,x]; df<-df[-x]; return(df)} #convert column to row.names
cumProb=function(x) 1-prod(1-x)  #define cumulative prob function
gg_colors <- function(n) {
		  hues = seq(15, 375, length=n+1)
		  hcl(h=hues, l=65, c=100)[1:n]
		} #color palette
###################################################
```

```{r loadData_etc, echo=F, error=F, message=FALSE, warning=F, include=FALSE}
##########LOAD METADATA############################
#Note: this project required merging data from two different sequencing runs, a MiSeq PE300 run and a MiSeq Nano PE300 run.  So below the code loads the metadata from each of those, then combines the two, then loads the combined OTU data.

#MiSeq full run metadata
  	meta=read.csv("MiSeq_metadata_wTides.csv")
  	sampleInfo<-read.csv("SampleCovariates.csv")
  		names(sampleInfo)[1]<-"original_sample"
  		meta<-merge(meta, sampleInfo, by="original_sample", all=T)
  		meta<-meta[1:99,] #get rid of extra empty rows
  	lib_tags= paste0("ID1=",meta$pri_index_name, ";ID2A=",meta$tag_sequence, ";ID2B=",as.character(reverseComplement(DNAStringSet(meta$tag_sequence))))  
  	  meta<-meta[-4,] #missing seq data for this sample
  	lib_tags<-lib_tags[-4]  #missing seq data for this sample
      meta$lib_tags<-lib_tags #add lib tags to metadata
      meta$run<-"20170727_MiSeq"
###get metadata from nano run, and merge
    metaNano=read.csv("MiSeq_nano_metadata.csv")
    #sampleInfo<-as.data.frame(gs_read(gs_title("Step1_Samples"))) #field data for covariates
    #names(sampleInfo)[1]<-"original_sample"
    metaNano <-merge(metaNano, sampleInfo, by="original_sample", all=T)
    metaNano<-metaNano[!is.na(metaNano$sample_name),]
    levels(metaNano$tag_sequence)<-c(levels(metaNano$tag_sequence), "noTag")
    metaNano$tag_sequence[metaNano$tag_number=="No_Tag"]<-"noTag"
      lib_tags_nano= paste0("ID1=", metaNano $pri_index_seq, ";", metaNano $tag_sequence)  
    metaNano$lib_tags<-lib_tags_nano
    metaNano$run<-"20170710_Nano"

#create merged metadatafile
allMeta<-merge(meta, metaNano, all=T)
    allMeta$sample_name<-as.character(allMeta$sample_name)
    allMeta<-allMeta[order(allMeta$sample_name),]
    #need to disambiguate sample names across runs and homogenize naming conventions
    allMeta$sample_name[74:81]<-paste0("TW20170311", gsub("TW11", "",allMeta$sample_name[74:81]))
    allMeta$sample_name[1:4]<-paste0("LL20170312", gsub("LL12", "",allMeta$sample_name[1:4]))
    allMeta$sample_name[34]<-"Ostrich4"
    allMeta<-allMeta[order(allMeta$sample_name),]
    
    #interpolate/harmonize salinity measurements [we measured with two different instruments -- a Hannah multi-probe (here, called "YSI", because that was shorter), and a hand-held analog refractometer. The refractometer proved more reliable and accurate.]
    mod<-lm(Salinity_YSI~Salinity_refractometer, data=allMeta)
    mod2<-lm(Salinity_refractometer ~Salinity_YSI, data=allMeta)
    	missingYSI<-which(is.na(allMeta$Salinity_YSI) &  !is.na(allMeta$Salinity_refractometer))
  	missingRefract<-which(is.na(allMeta$Salinity_refractometer) &  !is.na(allMeta$Salinity_YSI))  
    allMeta$Salinity_YSI[missingYSI]<-round((allMeta$Salinity_refractometer[missingYSI]*mod[[1]][2])+mod[[1]][1],1)
    allMeta$Salinity_refractometer[missingRefract]<-round((allMeta$Salinity_YSI[missingRefract]*mod2[[1]][2])+mod2[[1]][1],1)
#format date/time in metadata
 allMeta$DateTime<- as.POSIXct(strptime(paste(allMeta$Date, allMeta$Time), format="%Y%m%d %H:%M"))

 #load GPS coordinates
 
 #######################################################

##########LOAD OTU Data  ##############################
#OTU data
allOTUs<-
	read.table("OTU.map") %>%
	filter(V2%in%allMeta$lib_tags) 
  allOTUs$V2<-allMeta $sample_name[match(allOTUs$V2, allMeta$lib_tags)]
  allOTUs<-spread(allOTUs, key=V2, value=V3, fill=0)%>%
   Col2RN(1)

  READcount1<-sum(sum(allOTUs))
  OTUcount1<-nrow(allOTUs)
  
  ###First decontamination step: occupancy modeling analysis
load("ProbOccCOI_dataOUT08_22_46.Rdata")
  #filter OTUs by max occupancy prob; keeping those with at least 80% chance of being truly present, based upon distribution of occupancy probs
        a<-aggregate(dataOUT$probOcc, list(dataOUT$OTUname), max)
          a<-a[a$x>0.8,]  #take only OTUs for which at least one population has a high probability of being truly present.
        allOTUs<-allOTUs[match(a$Group.1, row.names(allOTUs)),]; rm(a)
	    allOTUs<-allOTUs[,-grep("LL20170312C.1", colnames(allOTUs), ignore.case=T)]#mislabeled sample
      allOTUs <-allOTUs[order(rowSums(allOTUs), decreasing=T),] #order by rowSums
	    allOTUs <-allOTUs[,order(colnames(allOTUs))] #order column names
	
	READcount2<-sum(sum(allOTUs))
	  OTUcount2<-nrow(allOTUs)

#for cumulative probability of individual Families	  
    load("OTUprob.RData")
    dataOUT<-OTUprob; rm(OTUprob)
	  
	  
#######################################################

					###################################################
					#interal DECONTAMINATION STEP using controls 
					###################################################
					samples_all <- allOTUs					
						CONTROL=allMeta[grepl("Ostrich", allMeta$sample_name),]  #select relevant samples from the metadata
						samples_con <- samples_all[,colnames(samples_all)%in%CONTROL$sample_name]  #control samples
						samples_env <- samples_all[,!colnames(samples_all)%in%colnames(samples_con)] #environmental samples
						control_taxon <- names(which.max(rowSums(samples_con)))					
						contam_proportion_df=data.frame(samples_con) 
						contam_proportion_df=scale(contam_proportion_df, center=F, scale=colSums(contam_proportion_df))  #for each dup/OTU in controls, calc frequency
							contam_proportion_vec =apply(contam_proportion_df, 1, max)  #take max of these frequencies.  this is the max proportional contribution of contamination to each of these OTUs/dups in the dataset. Accordingly, you will then subtract this proportion from each dup/OTU occurrence in the dataset.  A different (better?) way to do this would be to fit a likelihood distribution to the observed proportions of the dups/OTUs in the controls, and come up with the most likely proportion attributable to contamination.  But, in most cases, the most likely contribution is at or near zero (using a beta distrib), and in many cases the models don't converge anyway.
					
					#Step 1: remove the fraction of each OTU abundace attributable to contamination
					DECONTAM = samples_env
						for (i in 1:ncol(DECONTAM)){
							nreads=sum(DECONTAM[,i])
							contam_proportion_vec_forSample= ceiling(contam_proportion_vec*nreads)
							DECONTAM[,i]= samples_env[,i]-contam_proportion_vec_forSample
						}
					DECONTAM[DECONTAM <0]<-0  #assign zero to any reads for which there are a negative (corrected) number of reads
					DECONTAM = DECONTAM[rowSums(DECONTAM)>0,]  #remove reads that no longer occur in the data
					
					#Step 2: remove control taxon OTU from environmental samples in which it occurs
				  DECONTAM <- DECONTAM[!(rownames(DECONTAM) %in% control_taxon),]
					#remove any other OTU that predominantly occurs in control samples
					tail(samples_con[order(rowSums(samples_con)),], 20)  #visualize
			DECONTAM <- DECONTAM[!(rownames(DECONTAM) %in% row.names(tail(samples_con[order(rowSums(samples_con)),], 3))),] #here, 3 other OTUs... will differ by dataset
			DECONTAM <- DECONTAM[!(rownames(DECONTAM) %in% names(which(rowSums(samples_con)>rowSums(samples_env)))),]
	
			
	#auxiliary cleanup		
	DECONTAM<-DECONTAM[,colSums(DECONTAM)>10000] #low abundance outlier
	DECONTAM<-DECONTAM[,colSums(DECONTAM)<200000] #high abundance outlier
  DECONTAM<-DECONTAM[,-grep("LL20170311B.2", colnames(DECONTAM), ignore.case=T)]#low-abundance outlier
	DECONTAM <-DECONTAM[order(rowSums(DECONTAM), decreasing=T),]
	DECONTAM <-DECONTAM[,order(colnames(DECONTAM))]
	
	READcount3<-sum(sum(DECONTAM))
	  OTUcount3<-nrow(DECONTAM)

#######################################################
#Third decontamination: Outlier removal, for PCR replicates that are very different from their sister replicates
#######################################################
	replicates<-substr(colnames(DECONTAM),1,11)
	distmat<-vegdist(t(DECONTAM))
          distList=list(NA) ; distList.tri=list(NA); index=1
          for (i in unique(replicates)){
          	rowMatch<-which(replicates%in%i)
          	distList[[index]]<-as.matrix(distmat)[rowMatch, rowMatch]
          			distList.tri[[index]]<-	distList[[index]][upper.tri(	distList[[index]])]
          	index=index+1
          }

normparams=fitdistr(unlist(distList.tri), "normal")$estimate  #fit normal distribution to bray-curtis dissimilarities. lognormal, beta, etc, had less-good fits.
	probs=pnorm(unlist(distList.tri), normparams[1], normparams[2])
	outliers =which(probs>0.95)
	minOutlier<-min(unlist(distList.tri)[outliers]) #minimum outlier value

      #remove outliers
      distList<-distList[-which(lapply(distList, length)==1)] 	
      namesOutliers=list(NA)
      for (i in 1:length(distList)){
      	namesOutliers[[i]]<-intersect(
                                names(which(colSums(distList[[i]]>=minOutlier)>0)),
                                names(which.max(rowMeans(distList[[i]])))
                              )
      }
      DECONTAM<-DECONTAM[,-match(unlist(namesOutliers),colnames(DECONTAM))]
        DECONTAM<-DECONTAM[rowSums(DECONTAM)>0,]

        READcount4<-sum(sum(DECONTAM))
        	  OTUcount4<-nrow(DECONTAM)

#######################################################
#######################################################
	
#######################################################
##########LOAD Taxononmic Annotation Data  ############
tax<-read.csv("masterTaxonomyTable_20171012.csv", header=T, row.names=1)
    tax<-tax[order(tax$OTU_abundance, decreasing=T),]
#######################################################
#######################################################


####Load tidal data from NOAA
#got tidal data from https://tidesandcurrents.noaa.gov/noaatidepredictions.html
NOAA<-read.table("NOAAtideData_UnionMarch2017.txt", skip=13, header=T)
	NOAA$Time<-format(strptime(NOAA$Time, format="%H:%M"), "%H:%M")
	NOAA$Date<-as.Date(NOAA$Date)
	NOAA$DateTime<-as.POSIXct(strptime(paste(NOAA$Date, NOAA$Time, sep=" "), format="%Y-%m-%d %H:%M"))

##########Rarefy OTU Data  ##############################
#rarefy the data to make it a fair comparison between samples and replicates, etc.
set.seed(49)
rrOTUs<-
	DECONTAM%>%
	filter(rowSums(.)>0)%>%
	select(-grep("Ostrich", colnames(.), ignore.case=T))%>%
	t(.)%>%
	rrarefy(.,min(rowSums(.)))%>%
	t(.)
row.names(rrOTUs)<-gsub(";size=","",row.names(DECONTAM))	
	rrOTUs<-rrOTUs[rowSums(rrOTUs)>0,]
	rrMeta<-allMeta[match(colnames(rrOTUs), allMeta$sample_name),]   #metadata for columns included in rrOTUs
	row.names(rrOTUs)<-substr(row.names(rrOTUs),1,45) #truncate row.names to match those in the taxonomy database

	
  #prune taxonomic info to just OTUs present in rrOTUs
	rrTax<-tax[match(row.names(rrOTUs), row.names(tax)),]
#######################################################
#######################################################	
	
#merge df w tidal information
dataOUT$OTUname<-substr(dataOUT$OTUname, 1,45)    
OTUprob<-dataOUT[dataOUT$OTUname%in%row.names(rrOTUs),]
OTUprob<-data.frame(OTUprob, tax[match(OTUprob$OTUname, row.names(tax)),])
  
#create convenience variables
      Bottles<-substr(colnames(rrOTUs), 1,11)
      Sites<-substr(colnames(rrOTUs), 1,2)
      SeqRuns<-rrMeta$run
      Dates<-gsub("[A-Z]","", Bottles)
      Tides<-rrMeta$Tide
      Time<-format(strptime(rrMeta$time, format="%H:%M"), format="%H:%M")  #in hours (of a 24-hour clock)
      Sal<-rrMeta$Salinity_refractometer
      Temp<-as.numeric(rrMeta$Temperature)
      dna<-rrMeta$Qubit_ngul
      uniqueEvents <- expand.grid(c("LL.+", "PO.+", "TW.+"), c("11","12"), c("[ABC]", "[DEF]"))
      	uniqueEvents<-apply(uniqueEvents[,c("Var1", "Var2", "Var3")], 1, paste, collapse="") #create a vector to grep for each unique sampling event
          rrMeta$SamplingEvent<-NA #create col in metadata for sampling event ID
          for (i in 1:length(uniqueEvents)){
          	rrMeta $SamplingEvent[grep(uniqueEvents[i], colnames(rrOTUs))]<-LETTERS[i]
          }
      SamplingEvents<-rrMeta$SamplingEvent

#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
#######################################################
	
```

#Introduction

As environmental DNA (eDNA) becomes an increasingly important tool in ecological research (REFS), it is critical to understand how techniques for eDNA collection and analysis perform under real-world conditions (REFS). In particular, we must characterize the spatial and temporal resolution of amplicon-sequencing studies in order to confidently detect ecological patterns in the field (REFS). 

Most efforts to quantify the behavior of eDNA in the field have taken the form of quantitative PCR (qPCR) studies, in which the concentration of target template DNA -- either a particular target species or else a synthesized template not otherwise occurring in nature -- is measured over space or time. [Paragraph on these studies, w citations.]

By contrast, less work has focused on the behavior of eDNA as reflected in ecological amplicon-sequencing studies. [Cite/review these].

The changing tides of nearshore marine habitats are rigorous testing grounds for eDNA surveys, featuring large-scale movements of water every few hours (REFS). In addition, these nearshore habitats are among the most physically dynamic and biologically diverse on earth (REFS), such that eDNA as a survey technique holds particular promise for better understanding the thousands of species that may co-occur at a single location. 

Given recent work suggesting that eDNA signals are predominantly highly localized in space and time (REFS) -- although in some circumstances, eDNA may travel some distance -- we asked whether marine eDNA community composition changes over tidal cycles at a given location. A scenario in which eDNA communities change in unpredictable ways with each new tide would suggest an exogeneous origin for that DNA, such that DNA arrives at a site with incoming tides, drawn from a pool of organisms existing elsewhere. By contrast, consistent eDNA communities over multiple tidal cycles would strongly suggest an endogenous origin and highly localized signal. An alternative scenario is that high- and low-tide sigals at a site could differ in systematic and consistent ways, reflecting either highly endogeneous DNA (i.e., the species with which the water is in immediate contact at the time of sampling) [or else perhaps exogeneous origin that varies predictably with tide??]. 

In a spatially- and temporarily-replicated study, we find that nearshore COI eDNA community composition is not strongly influenced tide, and instead remains largely consistent within each geographic location across multiple successive tides. However, where dramatic shifts in the physical and chemical environment suggest arrival of a new water mass at a single sampling site, the eDNA community appears to change accordingly. Put differently, it seems likely that changes in aqueous habitat characteristics -- not tide itself -- yield changes in eukaryotic eDNA communities.

#Methods

##Field Sampling

```{r FIG1_sitemap, fig.width=4, fig.height=4, fig.path='figures/', dev=c('png', 'pdf'), echo=F, fig.cap="\\label{fig:fig1}Nearshore sampling locations in Hood Canal, Washington."}

lats<-c(47.462,47.377,47.378)
longs<-c(-123.113,-123.150,-122.973)
sitenames<-c("Lilliwaup","Potlatch","Twanoh")

### Get a map
map <- get_map(location = c(lon = mean(longs), lat = mean(lats)), zoom = 10,
               maptype = "terrain", source = "google")
plotmap<-ggmap(map)

plotmap + 
	geom_point(data=data.frame(lats,longs), aes(x=longs, y=lats), color="red", size=3) +
	geom_text(data=data.frame(lats,longs, sitenames), x=longs, y=lats-0.02, label=sitenames, fontface = "bold") +
	ggtitle("Sampling Locations" , subtitle="Hood Canal, Washington, USA") +
	theme(plot.title = element_text(hjust = 0.5), plot.subtitle = element_text(hjust = 0.5) ) +
	xlab("Longitude")+
	ylab("Latitude")


```

Our study design aimed to distinguish the effects of tide from site-level community differences and from sampling error. Consequently, we sampled each of three geographic locations (Fig 1; GPS coordinates given in Suppl. Table 1) four times -- twice during an incoming tide, and twice during an outgoing tide -- over a ca. 28-hour period. We collected three 1-L water samples for eDNA analysis (ca. 10m apart) at each site during each sampling event. Each sample was collected at the surface (< 0.5m depth), using a ca. 3m-long pole with plastic collection bottle attached. We kept samples on ice until they could be processed (within hours of collection). We filtered 500mL from each sample onto cellulose acetate filters (47mm diameter; 0.45um pore size) under vacuum pressure, and preserved the filter at room temperature in Longmire's buffer following Renshaw et al. (2015). Deionized water served as a negative control for filtering. We measured water temperature and salinity with a hand-held multiprobe (Hanna Instruments, Inc. model XXXX), as well as measuring salinity with a handheld manual refractometer.

  
```{r Table1_sampling_table, echo=F}
#xtabs(~Sites+Tides)
f<-as.data.frame(table(Bottles, Sites, Tides))
  f$Sites<-gsub("LL", "Lilliwaup", f$Sites)
  f$Sites<-gsub("PO", "Potlatch", f$Sites)
  f$Sites<-gsub("TW", "Twanoh", f$Sites)
  f$Freq[f$Freq>0]<-1  #convert to unique bottles of water sampled, rather than PCR replicates sequenced

  #print(xtable(xtabs(f$Freq~f$Sites+f$Tides)), comment=F)

  kable(xtabs(f$Freq~f$Sites+f$Tides), style="latex", align="c", caption="Samples by site and tide, showing balanced sampling design. Each site/tide combination (N = 6) had a total of 2 sampling events, consisting of 3 water samples per event, and then 3-4 PCR replicates per water sample, such that we sequenced 36-44 individual PCR replicates per geographic sampling site. Not all samples successfully amplified/sequenced; 93 individual replicates survived quality-control, described below.")
```


##DNA Extraction, Amplification, and Sequencing

We extracted total DNA from the filters using a phenol:chloroform:isoamyl alcohol protocol following [-@renshaw2015room], resuspended the eluate in 200uL water, and used 1uL of diluted DNA extract (1:10) as template for PCR. Although a single locus cannot completely characterize the biodiversity at a particular location [see, e.g., @kelly2017multilocus], we used a 330bp fragment of COI to assess the eukaryotic variance among our samples. This primer set [@leray_new_2013] amplifies a broad array of taxa including representative diatoms, dinoflagellates, metazoans, fungi, and others; here, we simply use this primer set as an assay to characterize community similarity among samples. We followed a two-step PCR protocol to first amplify and then index our samples for sequencing, such that we could sequence many samples on the same sequencing run while avoiding amplification bias due to index sequence [@odonnellPrimers].  PCR conditions were as follows#RGS: PCR mixes were 1X of HotStar Buffer, 2.5mM MgCl2, 0.5mM dNTP, 0.3uM of each primer and include 0.5 units of HotStar Taq (Qiagen) per 20 uL reaction. The first round of PCR consisted of 40 cycles, including a 'touchdown' approach from 62 to 46 (-1C per cycle), followed by 25 cycles at 46C. The indexing PCR used a similar protocol with only 10 cycles at 46C.

We generated three PCR replicates for each of 35 water samples (3 samples per sampling event, 4 sampling events per site, 3 sites = 36 water samples, of which 35 were processed successfully), and sequenced each replicate individually in order to assess the variance in detected eDNA communities due to stochasticity during amplification. We simultaneously sequenced positive (Ostrich (*Struthio camelus*) tissue, selected because of the absence of this species in our study sites) controls with identical replication. We carried negative controls through amplification, but did not sequence them, due to methodological issues associated with library preparation in samples without any discernable amplicon. No amplification was visible via gel elecrophoresis in the negative controls, and [qubit/bioanalyzer] analysis showed negligible amounts of DNA present in those samples after amplification. The positive controls provided us with consistent estimates of cross-contamination (see below), which we used in sequence quality-control prior to analysis.

Following library preparation according to manufacturers’ protocols (KAPA Biosystems, Wilmington, MA, USA; NEXTflex DNA barcodes, BIOO Scientific, Austin, TX, USA), sequencing was carried out on an Illumina MiSeq (250bp, paired-end) platform in two different batches: a MiSeq V.2 run and a MiSeq nano run.  These were processed separately through the first stages of bioinformatics analysis (see below), and then combined after primer removal for dereplication. PCR replicates (derived from the same sampled bottle of water) sequenced on different runs clustered together without exception (see Results), and thus combining the data from two sequencing runs was appropriate.

##Bioinformatics

We processed the resulting sequence reads with a custom Unix-based script [@banzai], which calls third-party programs (Mahé et al., 2015; Martin, 2011; Zhang et al., 2014) to move from raw sequence data to a quality-controlled dataset of counts of sequences from operational taxonomic units (OTUs). A total of `r READcount1` reads survived preliminary quality-control in the bioinformatics pipeline, representing `r OTUcount1` OTUs, most of which were rare (< 5 reads). We controlled for contamination in three ways, following our approach in [@kelly2017multilocus]. First, to address the question of whether rare OTUs are a function of low-level contamination or are true reflections of less-common amplicons, we used a site-occupancy model to estimate the probability of OTU occurrence (Royle and Link, 2006; Lahoz-Monfort et al. 2015), using multiple PCR replicates of each environmental sample as independent draws from a common binomial distribution. We eliminated from the dataset any OTU with <80% estimated probability of occurrence, yielding a dataset of `r READcount2` reads (`r OTUcount2` OTUs). We retained the occupancy-probability data for each OTU for later use as an alternative to simple presence-absence treatment of the final dataset (see below). Second, we estimated (and then minimized) the effect of potential cross-contamination among samples -- likely due to tag-jumping [CITE] or similar effects -- as follows: (1) we calculated the maximum proportional representation of each OTU across all control (here, ostrich) samples, considering these to be estimates of the proportional contribution of contamination to each OTU recovered from the field samples. (2) We then subtracted this proportion from the respective OTU in the field samples, yielding `r READcount3` reads (`r OTUcount3` OTUs). Finally, we dropped samples that had highly dissimilar PCR replicates (Bray-Curtis dissimilarities > `r round(minOutlier,2)`, which were outside of the 95% confidence interval given the best-fit model of the observed among-replicate dissimilarities). The result was a dataset of `r READcount4` reads (`r OTUcount4` OTUs), or `r (READcount4/READcount1)*100`% of the post-pipeline reads.

We rarefied read counts from each PCR replicate to allow for comparison across water samples using the vegan package for R (Oksanen et al., 2015). We carried out subsequent analyses on a single, illustrative rarefaction draw; these did not vary substantially among the rarefaction replicates (Supplemental Figure 1).

##Statistical Analysis

###Data Validation: Apportioning Variance in Bray-Curtis Dissimilarity Among Sites, Sampling Events, Bottle Samples, and PCR Replicates
We calculated the variance in OTU communities at four hierarchical levels -- among geographic sites (N = 3), among sampling events within geographic sites (N = 4 per site), among sample bottles within a sampling event (N = 3 per event per site), and among PCR replicates (N = 3 per individual sample bottle) -- using a PERMANOVA test on Bray-Curtis (OTU count data) dissimilarity among sequenced replicates. Calculations were carried out in R \cite{R_core} using the vegan \cite{} package. Having established that the variance among PCR replicates and bottles was small relative to variance among sampling events and geographic sites (see Results), it was clear that our dataset had the necessary resolution to detect community-level changes -- if any -- associated with changes in tide.

###How Many Ecological Communities Are Present, and How Similar are Communities Among Sites, Sampling Events, and Tides?

We then used Bray-Curtis dissimilarity to visualize differences among sampled communities at each hierarchical level of organization, using ordination (NMDS; R packages MASS \cite{} and ggplot2 \cite{}) and a heatmap. 

[Proposed solution: go w counts data for the main paper, and put probability/occupancy in a supplement. That way we can tell one story, and make reference to the fact the probOcc data also suggests that tides don't matter. ]

Given the strong and consistent differentiation we identified between two ecological communities in the eDNA data, we then labeled these communities 1 and 2, and applied a set of standard statistics to test for associations between community identity and geographic site (Fisher's exact test), tidal direction (incoming vs. outgoing; chi-squared), and tidal height (logistic regression).


###Community Identity by Site and Tide

We recovered tidal height data for our study sites during the relevant dates from the National Oceanographic and Atmospheric Administration data for Union, Washington (available at: https://tidesandcurrents.noaa.gov/noaatidepredictions.html).

###Characterizing the Observed Ecological Communities

We stress that a single genetic locus provides only a biased and incomplete view of an ecosystem (see \cite{Kelly_et_al_multilocus} for discussion), and although our purpose was to test for the effect of tidal fluctuations on detected eDNA communities -- which does not require taxonomic annotation of the recovered OTUs -- we were nevertheless interested in the membership of the ecological communities we detected. Our locus of choice, COI, provided a broad view of ecosystem... many phyla, trophic levels, etc.  [Supplmental table -- N OTUs assigned to various ranks.] 

We assigned a taxonomic name to each OTU sequence using blastn (Camacho et al., 2009) on a local version of the full NCBI nucleotide database (current as of August 2017), recovering up to 100 hits per query sequence and reconciling conflicts among equally good matches using the last common ancestor approach implemented in MEGAN 6.4 (Huson et al., 2011). 

`r 100*(sum(!is.na(rrTax$OTU_taxon_MEGAN))/nrow(rrOTUs))`% of OTUs could be annotated at some taxonomic level, with over half (`r 100*(sum(!is.na(rrTax$OTU_taxon_family))/nrow(rrOTUs))`%) being annotated to the level of taxonomic Family or lower.

We then report results for the most common XX OTUs, which account for the majority of the observed differences among ecological communities, with the full dataset being available as Supplemental Information.


#Results  

##Community-Level similarity among replicates, sites, etc: Apportioning variance in Bray-Curtis Dissimilarity

```{r create_distance_matrix, echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, results='hide'}

##Create distance matrices. Counts will use Bray-Curtis

OTU.count.distance<-as.matrix(vegdist(t(rrOTUs), method="bray", diag=F, upper=F, binary=F))

FamilyCounts<-  
  aggregate.data.frame(rrOTUs, list(tax$OTU_taxon_family[match(row.names(rrOTUs), row.names(tax))]), FUN=sum) %>%
  Col2RN(1)

OrderCounts<-  
  aggregate.data.frame(rrOTUs, list(tax$OTU_taxon_order[match(row.names(rrOTUs), row.names(tax))]), FUN=sum) %>%
  Col2RN(1)

ClassCounts<-  
  aggregate.data.frame(rrOTUs, list(tax$OTU_taxon_class[match(row.names(rrOTUs), row.names(tax))]), FUN=sum) %>%
  Col2RN(1)

Family.count.distance<-
  FamilyCounts%>%
  t()%>%
  vegdist(method="bray")%>%
  as.matrix()

#apportioning variance: OTU counts
invisible(
  OTU.count.adonis<-adonis(OTU.count.distance~rrMeta$Tide+rrMeta$Site.x+rrMeta$SamplingEvent+rrMeta$original_sample)$aov.tab
)

#Site-specific Tide analysis
OTU.count.by.site<-split.data.frame(OTU.count.distance, rrMeta$Site.y)
  LL<-OTU.count.by.site$Lilliwaup[, grep("LL", colnames(OTU.count.by.site$Lilliwaup))]
  PO<-OTU.count.by.site$Potlatch[, grep("PO", colnames(OTU.count.by.site$Potlatch))]
  TW<-OTU.count.by.site$Twanoh[, grep("TW", colnames(OTU.count.by.site$Twanoh))]

invisible(
LL.OTU.count.adonis<-adonis(LL~rrMeta$Tide[rrMeta$Site.y=="Lilliwaup"]+rrMeta$SamplingEvent[rrMeta$Site.y=="Lilliwaup"]+rrMeta$original_sample[rrMeta$Site.y=="Lilliwaup"])
)
invisible(
PO.OTU.count.adonis<-  adonis(PO~rrMeta$Tide[rrMeta$Site.y=="Potlatch"]+rrMeta$SamplingEvent[rrMeta$Site.y=="Potlatch"]+rrMeta$original_sample[rrMeta$Site.y=="Potlatch"])
)
invisible(
TW.OTU.count.adonis<-  adonis(TW~rrMeta$Tide[rrMeta$Site.y=="Twanoh"]+rrMeta$SamplingEvent[rrMeta$Site.y=="Twanoh"]+rrMeta$original_sample[rrMeta$Site.y=="Twanoh"])
)


##NMDS
	nmds.OTU.count<-metaMDS(OTU.count.distance, diag=T, upper.tri=T, trymax=1000, trace=0)
	NMDS.RESULTS<-data.frame(
	  OTU.count=scores(nmds.OTU.count),
	  Sample=row.names(scores(nmds.OTU.count)),
	  Bottles, SamplingEvents, Sites, SeqRuns, Dates, Tides, Sal, Temp
	  )
```	

To evaluate the spatial and temporal turnover between eDNA communities, we first apportioned the observed variation in COI Bray-Curtis dissimilarities (calculated using OTU read counts) among tides (incoming vs. outgoing), sampling sites, sampling events within a site, biological replicates (individual bottles of water taken during the same sampling event), and technical replicates (PCR replicates from the same bottle of water). Across the whole dataset, ecological communities at different sampling sites (20-50km apart) account for the largest fraction of the variance (`r OTU.count.adonis$R2[2]`), and different sampling events within those sites account for a similarly high proportion (`r OTU.count.adonis$R2[3]`). In contrast, biological replicates (N = 3 bottles of water per sampling event, taken ca. 10m apart) account for a small fraction (`r OTU.count.adonis$R2[4]`) of the variance, with differences among tides accounting for the smallest fraction of the variance in community dissimilarity (`r OTU.count.adonis$R2[1]`). The remainder -- `r OTU.count.adonis$aov.tab$R2[5]` -- is largely due to differences among technical PCR replicates (N = 3-5 per bottle of water), much of which derives from stochasticity in the presence of rare OTUs (Supplemental Figure 1). The comparitively low variance issuing from biological and technical replicates relative to sampling events and sites affords the resolution necessary to further examine questions of community composition across space and time. 

This approach to apportioning variance -- here, in Bray-Curtis dissimilarity -- depends upon the total variance in the dataset being analyzed. Analyzing individual site-level data eliminates the portion of variance due to between-site differences, effectively amplifying the contributions of the remaining hierarchical sampling levels. For each of our three sampling sites, differences among sampling events were greater than differences among tides.


```{r ADONIS_TreemapDiagrams, fig.width=6, fig.height=7.2, fig.path='figures/', dev=c('png', 'pdf'), echo=F, fig.cap="\\label{fig:fig2}Results of PERMANOVA, apportioning variance by hierarchical levels of sampling design: Tide (incoming vs. outgoing), Sampling Site, Sampling Event (N = 4 time points per site), and Sampling Bottle (N = 3 bottles per sampling event). Residuals reflect variance among PCR replicates (N = 3 replicates per sampling bottle) as well as variation due to rarefaction stochasticity and other sampling effects. The upper panel reflects results for the dataset as a whole, with lower panels giving site-specific variances. Numbers reflect proportion of the variance explained by the indicated hierarchical level (R$^2$), with p-values in parentheses."}

      q<-as.data.frame(OTU.count.adonis)
        row.names(q)<-c("Tide", "Site", "SamplingEvent", "Bottle", "Residuals", "Total")
        q<-data.frame(row.names(q), q)
        names(q)[1]<-"Level"
        q<-q[-which(row.names(q)=="Total"),]
        q$Level<-paste(q$Level, "\n",
                       round(q$R2,2), " (", q$Pr..F., ")", 
                       sep="")
  
q0.plot<-ggplot(q, aes(area = R2, label=Level))+
  geom_treemap(fill=brewer.pal(6, "Blues")[2:6], color=1, alpha=.8)+
  geom_treemap_text(colour = "black", place = "centre", grow = FALSE, size=10)+
  ggtitle("All Sites")+
  theme(aspect.ratio=1/1.618, plot.margin = margin(2, 2, -2, 2, "pt"), plot.background = element_blank())
  


       q1<-as.data.frame(LL.OTU.count.adonis$aov.tab)
         row.names(q1)<-c("Tide", "SamplingEvent", "Bottle", "Residuals", "Total")
           q1<-cbind(paste(row.names(q1), "\n",round(q1$R2,2)," (", q1[,6], ")", sep=""), 
                        q1)
         names(q1)[1]<-"Level"
         q1<-q1[-5,]
   

q1.plot<-ggplot(q1, aes(area = R2, label=Level))+
  geom_treemap(fill=brewer.pal(6, "Blues")[c(2,4:6)], color=1, alpha=.8)+
  geom_treemap_text(colour = "black", place = "centre", grow = FALSE, size=6)+
  ggtitle("Lilliwaup")+
  theme(aspect.ratio=1/1.618)
         

       q2<-as.data.frame(PO.OTU.count.adonis$aov.tab)
         row.names(q2)<-c("Tide", "SamplingEvent", "Bottle", "Residuals", "Total")
           q2<-cbind(paste(row.names(q2), "\n",round(q2$R2,2)," (", q2[,6], ")", sep=""), 
                        q2)
         names(q2)[1]<-"Level"
         q2<-q2[-5,]

q2.plot<-ggplot(q2, aes(area = R2, label=Level))+
  geom_treemap(fill=brewer.pal(6, "Blues")[c(2,4:6)], color=1, alpha=.8)+
  geom_treemap_text(colour = "black", place = "centre", grow = FALSE, size=6)+
  ggtitle("Potlatch")+
  theme(aspect.ratio=1/1.618)
         
         
         q3<-as.data.frame(TW.OTU.count.adonis$aov.tab)
           row.names(q3)<-c("Tide", "SamplingEvent", "Bottle", "Residuals", "Total")
             q3<-cbind(paste(row.names(q3), "\n",round(q3$R2,2)," (", q3[,6], ")", sep=""), 
                          q3)
           names(q3)[1]<-"Level"
           q3<-q3[-5,]

 q3.plot<-ggplot(q3, aes(area = R2, label=Level))+
  geom_treemap(fill=brewer.pal(6, "Blues")[c(2,4:6)], color=1, alpha=.8)+
  geom_treemap_text(colour = "black", place = "centre", grow = FALSE, size=6)+
  ggtitle("Twanoh")+
  theme(aspect.ratio=1/1.618)
          

 #do the actual plotting           
 multiplot(q0.plot, q1.plot, q2.plot, q3.plot,
           layout=matrix(c(1,2,1,3,1,4), nrow=2, ncol=3))     

  
```





#How Many Ecological Communities Are Present?


```{r merge_cofactor_data, echo=F,echo=FALSE,warning=FALSE,message=FALSE,error=FALSE}
#Label communities 1 (most of the samples) and 2 (Twanoh at two sampling events)

rrMeta$Community<-1
  rrMeta$Community[grep("TW20170311",rrMeta$sample_name)]<-2
  rrMeta$Community[grep("TW20170312[DEF]",rrMeta$sample_name)]<-2
  rrMeta$Community<-as.factor(rrMeta$Community)
  
  NOAA<-merge(
  rrMeta[,c("sample_name", "DateTime", "Community", "Salinity_refractometer", "Temperature", "Qubit_ngul")],
  NOAA, all=T)

 NOAA2<-NOAA[,-3]; names(NOAA2)[2]<-"bottle" ; NOAA2<-NOAA2[!is.na(NOAA2$bottle),];  NOAA2$bottle<-substr(NOAA2$bottle, 1, 11); NOAA2<-unique(NOAA2)
   OTUprob<-join(OTUprob, NOAA2, by="bottle", type="left")
   OTUprob$site<-substr(OTUprob$bottle, 1,2)

```



```{r multiplot_NMDS_BottleTide, fig.width=6, fig.height=7.2, fig.path='figures/', dev=c('png', 'pdf'), echo=F, fig.cap="\\label{fig:fig3}(A) Ordination plot (non-metric multidimensional scaling; NMDS) plot of Bray-Curtis dissimilarities among sequenced replicates, by sampling bottle (polygon) and tide (polygon color). (B) The same data shown as a heatmap. Only the Twanoh samples (upper right) stand out as having substantial heterogeneity, reflecting the two different communities present during different sampling events at that site."}     	
fig3a<-      ggplot() + 
      geom_polygon(data= NMDS.RESULTS,aes(x=OTU.count.NMDS1,y=OTU.count.NMDS2,fill=Tides,group=Bottles), alpha=0.3) +
      geom_text(data= NMDS.RESULTS,aes(x=OTU.count.NMDS1,y=OTU.count.NMDS2,label=Sites),alpha=0.4,size=2) +  # add the species labels
      #geom_point(data= NMDS.RESULTS,aes(x=OTU.count.NMDS1,y=OTU.count.NMDS2,colour= Tides),size=1.5) + # add the point markers
      scale_fill_manual(values = (c("navy", "skyblue"))) +
      #guides(fill=FALSE) +
      coord_equal() +
      theme_bw() +
      xlab("NMDS1") + ylab("NMDS2") +
      theme(axis.text.x = element_blank(),  # remove x-axis text
            axis.text.y = element_blank(), # remove y-axis text
            axis.ticks = element_blank(),  # remove axis ticks
            axis.title.x = element_text(size=12), # remove x-axis labels
            axis.title.y = element_text(size=12), # remove y-axis labels
            panel.background = element_blank(), 
            panel.grid.major = element_blank(),  #remove major-grid labels
            panel.grid.minor = element_blank(),  #remove minor-grid labels
            plot.background = element_blank(), 
            legend.position = "right",
            plot.title = element_text(hjust = 0.5),
            plot.margin = margin(0,0,20,0, unit="pt")
            ) +
      ggtitle("Bray-Curtis Dissimilarity\nby Sampling Bottle and by Tide")

      distmat<-as.matrix(vegdist(t(rrOTUs[,]), method="bray", diag=F, upper=F, binary=F))
      distmat<-melt(distmat)

    fig3b<-ggplot(distmat, aes(Var1, Var2, fill=as.numeric(value)))+
          geom_tile()+
          xlab("Sample")+ylab("Sample")+
          guides(fill=guide_legend("Bray-Curtis\nDissimilarity"))+
          theme(axis.text.x = element_text(angle=90, size=6),
            axis.text.y = element_text(size=6),
            panel.background = element_blank(),
            plot.background = element_blank())
  
  #heatmap(distmat, symm=T, labRow = Sites, labCol=Sites) #labRow=rrMeta$run, labCol=rrMeta$run

multiplot(fig3a, fig3b)

```

An ordination plot of Bray-Curtis distances among each of our xx sequenced replicates (Figure 2a) distinguishes two clusters of eDNA communities present across all samples. In agreement with the analysis of variance, technical PCR replicates and biological replicates consistently cluster closely in ordination space, yet two non-overlapping eDNA sequence assemblages appear on this plot. A heatmap of Bray-Curtis distances between all replicates reveals the underlying magnitudes of dissimilarity and clustering, showing two clearly distinct communities of eDNA sequences (Fig 2b). The two observed sequence clusters are primarily associated with sampling site. Specifically, the right-hand community is only present in three of four Twanoh samples, whereas the left-hand community is present in all technical and environmental replicates of a single Twanoh sample, as well as all Lilliwaup and Potlatch samples.



#Community Identity by Site and Tide


```{r tide_community_figure, fig.path='figures/', dev=c('png', 'pdf'), echo=F,echo=FALSE,warning=FALSE,message=FALSE,error=FALSE, fig.cap="\\label{fig:fig4}eDNA Communities by Time, Tide, and Site. Here we identify community type 1 as the dominant eDNA community (as seen in Figure 3, which appears at every geographic site), and community type 2 as the distinct type occurring only at Twanoh."}


#plot w ggplot
    ggplot(data=NOAA[130:350,], aes(x=DateTime, y=Pred))+
      ylab("Tidal Height (m)")+
      xlab("Day and Time")+
      theme_bw()+
      geom_smooth(se=F, span=.22, na.rm=F, color="grey")+
      geom_point(aes(shape=substr(sample_name,1,2), color=Community), data=NOAA[!is.na(NOAA$sample_name),], size=4)+
      labs(shape="Site", color="Community Type")
```

To examine the relationship of each eDNA community with tide, we first visually examined the mapping of tidal direction onto our ordination analysis (Figure 2a) and plotted community membership of each sample across the tidal cycle during collection (Figure 2b). Both qualitatively indicate a lack of association between tidal direction or height and the two eDNA communities. Quantitatively, by sampling event, it is clear that community is independent of tidal height (p = `r round(summary(lm(Pred~Community, data=NOAA))$coeff[8], 3)`; linear regression) and of tidal direction (incoming vs. outgoing; p = `r round(chisq.test(x=rrMeta$Tide, y=rrMeta$Community)$p.value, 3)`; $\chi^2$), but is related imperfectly to site identity (p = `r fisher.test(rrMeta$Site.x, rrMeta$Community)$p.value`; Fisher's exact test), as suggested previously (Figure 2). The single Twanoh sample belonging to eDNA Community 1 suggests that geography does not fully explain differences between these communities, and that ecological variables warrant further investigation as driving differences in communities.  

<!-- Information below = table 2? -->
```{r factors_predicting_community, warning=F, error=F} 
#anova(glm(Community~Temperature+Salinity_refractometer+Qubit_ngul+Tide, data=rrMeta, family="binomial"))

kable(as.data.frame(anova(
  glm(Community~as.numeric(Salinity_refractometer)*as.numeric(Temperature), data=rrMeta, family="binomial", na.action = na.omit))), format="latex", align="c", caption="The salinity and temperature of the sampled water mass explain nearly all of the variation in community identity across sites")

#stepAIC(glm(Community~Temperature+Salinity_refractometer+Qubit_ngul, data=rrMeta, family="binomial"))
```


[Note: insert model testing, etc. here]

To identify ecological factors that might distinguish the two eDNA assemblages observed <!-- I don't want to call these ecological communities until we have some basis upon which to do so... unless OTU similarity is good enough?--> (Figures 1 & 2), we examined the association of each sample's temperature, salinity, DNA extraction concentration (a proxy for primary productivity), and site identity <!-- could we put the site ID into these models, and show that it performs worse as a predictor of community membership than temp?--> with communities 1 and 2 (Table X). Our model demonstrates that temperature is the best <!--only valid?--> predictor of community membership, and that salinity, DNA extraction concentration, and site identity <!-- true? --> do not make significant contributions.  In summary, the eDNA communities are more closely associated with a single ecological variable, temperature, than with tide, or even with geographical origin. This suggests that the two eDNA assemblages may represent different aqueous habitat types, and led us to investigate their taxonomic composition.
<!-- Save the following for conclusion - no one knows it's plankton yet: "This, in turn, suggests that communities (particularly planktonic communities) change when the water characteristics change." -->

#Which organisms make up communities 1 and 2?

###FIGURE multiplot, community membership and covariates
```{r multiplot_community_membership, fig.width=6, fig.height=7.2, fig.path='figures/', dev=c('png', 'pdf'), echo=F}
  #################
    NOAA2<-
      NOAA%>%
      rename(bottle = sample_name)%>%
      filter(!is.na(bottle))%>%  
      mutate(bottle = substr(bottle, 1, 11), Temperature = as.numeric(Temperature))%>%
      unique()%>%
      mutate(BottleOrder = seq(1, nrow(.), 1))

    FamilyCountsSums<-
      aggregate.data.frame(t(FamilyCounts), list(substr(colnames(FamilyCounts),1,11)), FUN=sum)%>%
      Col2RN(1)%>%
      t()
    #FamilyCountsSums<-FamilyCountsSums[,-2]  # LL20170311B is unreplicated; remove

        
s1<-ggplot(aes(y=Pred, x=BottleOrder, color=Community, group=Pred), data=NOAA2)+geom_line(size=2)+xlab("")+ylab("Tide Height")+guides(color=F)+
    coord_cartesian(xlim=c(2, 34))+theme_bw()+theme(axis.text.x = element_blank())+theme(plot.margin = unit(c(0,5,-5.5,5), "pt"))+theme(axis.title=element_text(size=8))

s2<-ggplot(aes(y=Temperature, x=BottleOrder, color=Community, group=Pred), data=NOAA2[!is.na(NOAA2$Temperature),])+geom_line(size=2)+xlab("")+ylab("Temperature")+guides(color=F)+coord_cartesian(xlim=c(2, 34))+theme_bw()+theme(axis.text.x = element_blank())+theme(plot.margin = unit(c(-5,5,-5.5,5), "pt"))+theme(axis.title=element_text(size=8))

s3<-ggplot(aes(y=Qubit_ngul, x=BottleOrder, color=Community), data=NOAA2)+geom_point(size=2)+xlab("")+ylab("Total DNA Recovered")+guides(color=F)+
    coord_cartesian(xlim=c(2, 34))+theme_bw()+theme(axis.text.x = element_blank())+theme(plot.margin = unit(c(-5,5,-5.5,8), "pt"))+theme(axis.title=element_text(size=8))

s4<-ggplot(aes(y=Salinity_refractometer, x=BottleOrder, color=Community, group=Pred), data=NOAA2)+geom_line(size=2)+xlab("")+ylab("Salinity")+guides(color=F)+
    coord_cartesian(xlim=c(2, 34))+theme_bw()+theme(axis.text.x = element_blank())+theme(plot.margin = unit(c(-5,5,-12,5), "pt"))+theme(axis.title=element_text(size=8))

  
z<-FamilyCountsSums
  z<-z[order(rowSums(z), decreasing=T),]
  z<-z[1:10,]; z<-sweep(z, 2, colSums(z), "/")
  z<-data.frame(Family=row.names(z), z)
  z<-melt(z, id="Family")
  z$variable<-factor(z$variable, levels=unique(NOAA2$bottle))
s5<-ggplot(z, aes(y=value, x=variable, fill=Family))+
    geom_bar(stat='identity')+theme_minimal()+scale_fill_brewer(palette="Spectral")+
    theme(axis.text.x = element_text(angle = 90, hjust = 1), legend.position = "bottom")+theme(axis.title=element_text(size=8))+
    guides(fill=guide_legend(ncol=5, size=6))+xlab("Sample")+ylab("Proportion of Reads, Top 10 Families")+theme(plot.margin = unit(c(0,0,0,-2), "pt"))

multiplot(s1,s2,s3,s4,s5, layout=matrix(c(1,2,3,4,5,5,5), nrow=7, ncol=1))     

```


[Yeah, let's fix this together.]
<!-- To identify the taxonomic groups that most strongly differentiate ecological communities 1 and 2, we first performed principal component analysis on the OTU counts <!--better way to describe this? And is it actually OTU counts or families? for each replicate in our study, and examined the highest loadings on the first and second axes. We found that PCA separates our samples into two communities (Figure X) identical to those discovered by NMDS ordination (Figure 2), and that these groups can be distinguished primarily by the frequency of sequences from X genera <!-- move to family level?-->, including Bathycoccus (green algae) and Micromonas (green algae), Phaeocystis (green algae), Barantolla (polychaete worm with larval phase), Balanus (barnacle with larval phase), Minutocellus (algae), Centropages (copepod), Homo (terrestrial mammal), and Dictyocha (photosynthetic algae) (Table X <!--with read counts for each of these genera/families in each community, as well as common name and life history info so we don't need to put it in the text?-->). Because PCR amplification with universal primers may contribute to bias in relative OTU counts across taxonomic groups, we also performed PCA on the results of site occupancy modelling for each family identified in a replicate of our study. This approach effectively increases the contribution to PCA of families for which small numbers of sequence reads appeared reliably in multiple replicates and samples <!-- am I understanding site occupancy modelling yet?--> With this approach, we see a similar/different result...<!--seems likely we'll get a sim result, as the higher the read counts, the higher the site occupancy probability, on average right?--> (Table Y <!--or continuation of Table X-->). A handful of common green algae and benthic invertebrates (each of which have larval phases) therefore distinguish Communities 1 and 2; however, it remains unclear if these groups differ additionally in their overall abundance of OTUs.


<!---
To determine x (broadly), we did y. Specifically, we found z. Generally, without getting too conclusiony, this suggests w, and transitions into the next point. 
-->
<!---To quantify richness and diversity in Communities 1 and 2 during both incoming and outgoing tides, we examined the total number of unique OTUs and beta diversity from each replicate/sample. <!--There's data missing from this plot, I think. Potlatch has none for incoming community 1, and I believe it should, according to the tides plot. Also, would you consider making both plots in both forms (separated by tide/site AND collapsed? Then we could examine the role of tides in both, AND do some nonparametric comparisons once collapsed) --> No clear trends in species richness occur by community identity and tidal direction, though Community 2 (found only at Twanoh) appears to have greater variance among replicates <!--interesting enough to perform fligner.test(x, ...) to examine this possibility?-->.  Beta diversity is significantly <!--(?)--> higher in Community 2, despite the smaller number of samples and narrower geographic bounds on this group of samples. <!--transition sentence-->




```{r BrayCurtis_timeSeries, fig.width=6, fig.height=7.2, fig.path='figures/', dev=c('png', 'pdf'), echo=F, warning=F, message=F, fig.cap="\\label{fig:fig4}Comparison of Bray-Curtis dissimilarities within a reference sampling event (Time step = 0) and between the reference sample and subsequent samples at the same site (Time Steps 1, 2, and 3). Subsequent time steps reflect the accumulation of ecological eDNA differences over hours as the tide moves in and out. Sites shown individually. Note the relatively small accumulation of Bray-Curtis dissimilarity across time and tides with the exception of the Twanoh event described elsewhere, which shifts the community significantly. Y-axes identical to facilitate comparison across sites."}
#bray-curtis timeseries, within sites
  
LL.sampling<-rrMeta$SamplingEvent[rrMeta$Site.x=="Lilliwaup"]; diag(LL)<-NA

      BClist<-list()
      for (i in 1:4){
        BClist[[i]]<-LL[1:4,which(LL.sampling==unique(LL.sampling)[i])]
      }
      LL.event1<-lapply(BClist, as.vector)
      
      LL.event1<-data.frame(value=unlist(LL.event1),
                 timestep=c(rep(0, times=length(LL.event1[[1]])),
                               rep(1, times=length(LL.event1[[2]])),
                               rep(2, times=length(LL.event1[[3]])),
                              rep(3, times=length(LL.event1[[4]]))
                            ),
                 eventReferenceNumber=rep(1, times=length(unlist(LL.event1)))
                 )
      LL.event1<-LL.event1[!is.na(LL.event1$value),]

LL.plot<-ggplot(data= LL.event1, 
       aes(y=value, x=factor(timestep)))+
        geom_boxplot(fill="lightblue", alpha=0.5)+
        ggtitle("Lilliwaup")+
        #scale_x_discrete(rrMeta$Tide[rrMeta$Site.x=="Lilliwaup"], breaks=c(0,1,2,3), labels=c("Incoming", "Outgoing", "Outgoing", "Incoming"))+
        xlab("Time Step") + ylab("Bray-Curtis")+ylim(0,0.8)+theme_bw()

#TODO -- add secondary x-axis to reflect time-since-reference, and a tertiary axis to reflect tidal direction

################
#POTLATCH

PO.sampling<-rrMeta$SamplingEvent[rrMeta$Site.x=="Potlatch"]; diag(PO)<-NA

BClist<-list()
for (i in 1:4){
  BClist[[i]]<-PO[1:4,which(PO.sampling==unique(PO.sampling)[i])]
}
PO.event1<-lapply(BClist, as.vector)

          PO.event1<-data.frame(value=unlist(PO.event1),
                                timestep=c(rep(0, times=length(PO.event1[[1]])),
                                           rep(1, times=length(PO.event1[[2]])),
                                           rep(2, times=length(PO.event1[[3]])),
                                           rep(3, times=length(PO.event1[[4]]))
                                ),
                                eventReferenceNumber=rep(1, times=length(unlist(PO.event1)))
          )
          PO.event1<-PO.event1[!is.na(PO.event1$value),]

PO.plot<-ggplot(data= PO.event1, 
                aes(y=value, x=factor(timestep)))+
  geom_boxplot(fill="lightblue", alpha=0.5)+
  ggtitle("Potlatch")+
  xlab("Time Step") + ylab("Bray-Curtis")+ylim(0,0.8)+theme_bw()

##########
###TWANOH
TW.sampling<-rrMeta$SamplingEvent[rrMeta$Site.x=="Twanoh"]
diag(TW)<-NA
TW.sampling<-rrMeta$SamplingEvent[rrMeta$Site.x=="Twanoh"]; diag(TW)<-NA

BClist<-list()
for (i in 1:4){
  BClist[[i]]<-TW[1:4,which(TW.sampling==unique(TW.sampling)[i])]
}
TW.event1<-lapply(BClist, as.vector)

        TW.event1<-data.frame(value=unlist(TW.event1),
                              timestep=c(rep(0, times=length(TW.event1[[1]])),
                                         rep(1, times=length(TW.event1[[2]])),
                                         rep(2, times=length(TW.event1[[3]])),
                                         rep(3, times=length(TW.event1[[4]]))
                              ),
                              eventReferenceNumber=rep(1, times=length(unlist(TW.event1)))
        )
        TW.event1<-TW.event1[!is.na(TW.event1$value),]

TW.plot<-ggplot(data= TW.event1, 
                aes(y=value, x=factor(timestep)))+
  geom_boxplot(fill="lightblue", alpha=0.5)+
  ggtitle("Twanoh")+
  xlab("Time Step") + ylab("Bray-Curtis")+ylim(0,0.8)+theme_bw()



#####Multiplot
#pdf("/Users/rpk/GoogleDrive/Kelly_Lab/Projects/TideEffectHoodCanal_Gallego/Analysis/figures/TimeSeries_BC.pdf")
multiplot(LL.plot, PO.plot, TW.plot)
#dev.off()
#TODO -- stats



```



##Is there less turnover, tide-to-tide, among sessile species? 
Zoom in on Twanoh, in transitions between communities

<!---
To determine x (broadly), we did y. Specifically, we found z. Generally, without getting too conclusiony, this suggests w, and transitions into the next point. 
-->

To examine the turnover between ecological communities that occured in the span of X hours at a single site (despite previous directional shifts in tide at this location not resulting in a community change-over) changing communities), we focused further on the community composition, richness, and diversity of the two samples taken at Twanoh on March 12 at 11:00 (Community 1) and 13:00 (Community 2)...  

```{r TW_Order_turnover, fig.path='figures/', dev=c('png', 'pdf'), echo=F, warning=F, message=F, fig.cap="\\label{fig:fig5}Most influential OTUs, plotted by taxonomic Order, distinguishing the two ecological communities observed in water samples from Twanoh State Park. Shown are the taxonomic Orders of OTUs with at least 1000 reads in the rarefied dataset, and having a constrained canonical correspondence analysis (CCA) score of greater than 0.7 (absolute value), which in our dataset most clearly divides the two communities. See text for CCA details. The vertical black lines in the chart delineate the communities identified previously by NMDS (see Figure 3a), with the time of each sample given along the x-axis, showing the shift from one community to the other -- and then largely back again -- within less than 24 hours. Note that each block of samples reflects a different point in the tidal cycle, and that the first two time points indicate a continuity of community membership despite a change in tide (see Figure 4)."}
#just TW
  TW.OTUs<-rrOTUs[,grepl("TW", colnames(rrOTUs))]; TW.OTUs<-TW.OTUs[rowSums(TW.OTUs)>0,]
z<-cca(t(TW.OTUs)~rrMeta$Community[rrMeta$Site.x=="Twanoh"])

#plot(data.frame(z$CCA$v, rowSums(TW.OTUs))) #visualize influential data points
InfluentialOTUs<-TW.OTUs[which(rowSums(TW.OTUs)>1000&abs(z$CCA$v)>.7),]
InfluentialOTUs<-aggregate.data.frame(InfluentialOTUs, list(tax[row.names(InfluentialOTUs),"OTU_taxon_order"]), FUN=sum)
#InfluentialOTUs<-InfluentialOTUs[-match(c("No hits", "Not assigned"), InfluentialOTUs[,1]),]
InfluentialOTUs<-Col2RN(InfluentialOTUs, 1)
InfluentialOTUs<-aggregate(t(InfluentialOTUs), list(substr(colnames(InfluentialOTUs),1,11)), sum)  #aggregate by bottle
  InfluentialOTUs<-t(Col2RN(InfluentialOTUs, 1))

InfluentialOTUsMelted<- melt(as.matrix(InfluentialOTUs))

ggplot(InfluentialOTUsMelted[grepl("TW", InfluentialOTUsMelted$Var2),], aes(x=Var2, y=Var1, fill=log(value)))+
  geom_tile()+
  theme_bw()+
  scale_fill_gradient(low = "white",high = "steelblue", name="Log(Reads)")+
  ggtitle("Twanoh, Most Influential OTUs by Taxon")+
  scale_y_discrete("OTU by Taxon", labels=row.names(InfluentialOTUs), expand=c(0,0))+
  scale_x_discrete("Sample Date and Time", labels=format(rrMeta$DateTime[rrMeta$Site.x=="Twanoh"], "%Y-%m-%d %H:%M"), expand=c(0,0))+
  geom_vline(aes(xintercept=6.5),colour="black")+
  geom_vline(aes(xintercept=9.5),colour="black")+
  theme(axis.line = element_line(colour = "black"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5, margin = margin(t = 10, b = 15)),
        legend.position="bottom",
        axis.text.x = element_text(angle = 90, hjust = 1, size=6))

```

```{r taxon_attribute_table, echo=F, warning=F, message=F}

taxonAttributes<-data.frame(
   Order=row.names(InfluentialOTUs),
   Kingdom=c("Animalia", "Heterokonta", "Heterokonta", "Plantae", "Heterokonta", "Dinoflagellata","Oomycota", "Animalia", "Animalia"),
   Mobility=c("Motile", "Planktonic", "Sessile", "Planktonic", "Planktonic", "Planktonic", "Unknown", "Sessile", "Sessile"),
   Habitat=c("Marine", "Marine", "Marine", "Marine", "Marine", "Marine", "Terrestrial/Aquatic", "Marine", "Marine"),
   Cellularity=c("Multicellular", "Single-celled", "Multicellular", "Single-celled", "Single-celled", "Single-celled", "Single-celled", "Multicellular", "Multicellular")
)

kable(taxonAttributes, format="latex", align="c")


```




#Discussion

Environmental DNA is rapidly becoming an essential and widely-used tool to identify community membership in aquatic environments (review REFS). It is not yet clear to what extent the sequences identified in eDNA studies reflect the presence of local organisms in time and space, however (locality REFS). Of particular interest in marine systems is the influence of tide on the detection of ecological communities: Must sampling schemes standardize tidal height and direction during collection to detect consistent groups of species? Does each tide bring with it a new water masse, carrying exogenous DNA, or do the sequences detected at any given time accurately reflect the species present within a habitat in that moment? To address these questions, we collected and analyzed eDNA communities at three different sites along the Hood Canal over the course of two tidal turnovers (Figure X - sampling scheme). Thus, for each site, we were able to examine the influence of reversals in tidal direction on two separate occasions.  

When analyzed together, eDNA collections from three locations (Lilliwaup, Potlatch, and Twanoh) show substantial variance in OTU membership and prevalence associated primarily with geographic location (Adonis table/fig). Grouping of samples in ordination space is also strongly associated with site, rather than with tide (all-sample NMDS). Together, these results suggest that eDNA surveys designed to clarify relationships between distinct ecological communities are not likely to suffer substantially from sample collection at varying points in the tidal cycle, because the twice-daily exchange (or at least, mass movement) of water into- and out of our sampling sites appeared to have little influence on the sequences detected overall. 

Although the effect of tide on eDNA community composition is small when multiple geographic sites are considered simultaneously, tidal direction may still strongly influence the OTUs detected within a single location. The existence of among-site differences in ecological community in fact provide the resolution necessary to detect such a local influence of tide, if present - exogenous DNA arriving periodically with tidal flow at each site might closely resemble neighboring communities, and differ consistently from endogenous DNA collected on the ebb tide, which has spent hours in contact with local benthic flora and fauna. A site-by-site analysis reveals that a substantial proportion of the variance in OTU counts is associated with tide, but never as much as is associated with sampling event (Adonis single site table/figs). Additionally, visualization of samples from each geographic location in ordination space shows no recognizable grouping by tide (NMDS single site figs). Finally, the eDNA community present at a single site tends to drift over time, decreasing in similarity to the initial sample, regardless of tide (Time graphs). Together, these results suggest that the effect of tidal flow on eDNA community membership increases when geographic sites are considered independently, but remains small in comparison to the importance of sampling event, overall.  

Rather than tide, ecological variables such as temperature and salinity, each of which differ among sites and sampling events, seem to drive the bulk of the variance in eDNA community membership (Stacked Figure). These variables are key determinants of marine habitat type, especially for planktonic species. At Twanoh we sampled by chance a dramatic turnover in species composition within the span of just a few hours, and a parallel shift in ecological variables. The families most notably associated with this shift in community are indeed planktonic, suggesting that eDNA survey methodology succeeded in identifying the species physically present within the water column at the time of sampling. However, the entrance and exit of a water masse and planktonic community with characteristics more common at neighboring sites decreased but did not eradicate the signal from sessile species that consistently appeared at Twanoh. In summary, the sequenced eDNA community reflects contributions from both organisms living within the water itself, as well as sessile species in contact with that more mobile community.

Taken together, our results suggest that eDNA samples taken from even highly dynamic environments reflect recent contributions from local species. With the exception of the occasional movement of water masses representing distinct habitats for planktonic organisms, the eDNA communities we sampled at three geographic sites were largely stable over time and tide.  Practically, this suggests that intertidal eDNA research should be performed with substantial attention to ecological variables such as temperature and salinity, which serve as markers of the aqueous habitat present, and may not remain consistent geographically. In contrast, tidal turnover appears to be a secondary consideration that does not dramatically or consistently affect the commmunity sampled, even within a single geographic location. Marine intertidal eDNA surveys therefore appear to reflect the endogenous DNA of the organisms present in the water and on the benthic substrate at the time of sampling.



<!--
Where nearby habitats share the same complement of species, it becomes difficult to distinguish DNA of exogenous versus endogenous origin,  because of the lack of contrast between the two.  By chance our sampling design detected starkly different ecological communities associated with  different water masses, allowing us to identify a changeover between communities at one of our sampling sites, while the communities at the other two sampling sites remained constant over multiple tides. Taken together, these results suggest that eDNA samples taken from even highly dynamic environments reflect recent contributions from local species.

* 
* These results suggest that eDNA communities are likely to be stable over time, consistent with the stability of the sampled habitats.
* By contrast, where water characteristics -- as opposed to mere tidal fluctuations having the same water characteristics -- changed, we detected a wholly different eDNA community. 
* This change was mainly visible in the planktonic species, and in particular, in marine microalgal species, although this may be because of these species' being most prominently represented in our 18S dataset. 
* Our eDNA samples appear to reflect the immediate aquatic habitat sampled, rather than exogeneous DNA from elsewhere; this is consistent with earlier studies. 
-->

  


##Acknowledgements

##Funding

#References




#Supplemental Information
```{r Supplement_GPScoordinatesSamplingAreas, echo=F, message=F}

kable(data.frame(
  Site=c("Lilliwaup State Park","Potlatch State Park","Twanoh State Park"),
  Latitude=c(47.462,47.377,47.378),
  Longitide=c(-123.113,-123.150,-122.973)
), format="latex", align="c", caption="Supplemental Table 1: Sampling Sites")
```



```{r Supplement_Rarefaction, fig.width=4, fig.height=6, fig.path='figures/', dev=c('png', 'pdf'), echo=F, message=F}

#Note: the loop below takes a minute or two to process, so I have saved output to save time.
# rrSupplement<-list()
# for (i in 1:100){
# rrSupplement[[i]]<-
# 	DECONTAM%>%
# 	t(.)%>%
# 	rrarefy(.,min(rowSums(.)))%>%
#   vegdist(method="bray")%>%
#   as.vector()
# }
#save(rrSupplement, file="rrSupplement_rarefactionDraws.Rdata")
load("rrSupplement_rarefactionDraws.Rdata")
hist(unlist(lapply(rrSupplement, median)), xlim=c(0.3,0.5), main="Median Bray-Curtis Dissimilarity\nEach of 100 Rarefaction Draws", xlab="Bray-Curtis Dissimilarity")

```



###SUPPLEMENTAL FIGURE: Looking at that stochastic variation among technical replicates:Stochasticity dominates after the first few percentiles (i.e., the first few hundred OTUs). This is interesting, but expected. [Jaccard/binary looks the same, but slightly elevated, as you'd expect.]
```{r stochastic_variation_rareTail, echo=FALSE,warning=FALSE,message=FALSE, error=FALSE,  fig.height = 4, fig.width = 5}
		binNo<-cut(1:nrow(rrOTUs), 100, labels=F)  #N equal-sized groups
		MeanBC<-list()
		replicates=substr(colnames(rrOTUs), 1,11) #create vector of replicate names
		for (j in 1:length(unique(binNo))){
			distmat=vegdist(t(rrOTUs[binNo==j,]), upper=T, diag=T) #calculate bray-curtis distances
			distList=list(NA) ; distList.tri=list(NA); index=1  #create empty lists to fill in, and then create list of matrices, each of which is distances among replicates
				index=1
				for (i in unique(replicates)){
					rowMatch<-replicates%in%i
					distList[[index]]<-as.matrix(distmat)[rowMatch, rowMatch]
							distList.tri[[index]]<-	distList[[index]][upper.tri(	distList[[index]])]
					index=index+1
				}
			MeanBC[[j]]<-unlist(lapply(distList, mean))
		}
		outBC<-unlist(MeanBC)
		xdata<-rep(1:length(unique(binNo)), each=length(outBC)/length(unique(binNo)))

		#ggplot(aes(y=outBC[1:1750], x=xdata[1:1750]))+geom_boxplot()
		
		
		boxplot(outBC[1:1750]~xdata[1:1750],col="skyblue", main="Mean Bray-Curtis Among Technical Replicates, by OTU Commonnness", xlab="Percentile OTU Commonness", ylab="Mean Bray-Curtis Among Technical Replicates", las=2)

```